src/directives/formly-form.test.js

```diff
@@ -933,12 +933,19 @@ describe('formly-form', () => {
 
       const listener = sinon.spy()
       const expression = sinon.spy()
+      const deepLinkField = getNewField()
+      deepLinkField.key = 'foo.bar'
+      deepLinkField.watcher = {
+        listener: '',
```

Comment:
Before I recommit.

Does this work for you?

it(`should setup any watchers specified on a field`, () => {
      scope.model = {}

      const listener = sinon.spy()
      const expression = sinon.spy()
      const deepLinkField = getNewField()
      deepLinkField.key = 'foo.bar'
      deepLinkField.watcher = {
        listener: sinon.spy(),
        expression: sinon.spy(),
      }

      scope.fields = [getNewField({
        watcher: {
          listener: '',
        },
      }),
      deepLinkField, getNewField({
        watcher: [{
          listener: '',
          expression: '',
        }, {
          listener,
          expression,
        }],
      })]

      expect(compileAndDigest).to.not.throw()
      expect(listener).to.have.been.called
      expect(expression).to.have.been.called
      expect(deepLinkField.watcher.listener).to.have.been.called
      expect(deepLinkField.watcher.expression).to.have.been.called
    })
  })

From:  "Kent C. Dodds" <notifications@github.com>
Reply-To:  formly-js/angular-formly <reply@reply.github.com>
Date:  Monday, November 9, 2015 at 5:24 PM
To:  formly-js/angular-formly <angular-formly@noreply.github.com>
Cc:  Matt Kruczek <kruz79@gmail.com>
Subject:  Re: [angular-formly] fix(formly-form): Issue number 542 (#545)


In src/directives/formly-form.test.js:
> @@ -933,12 +933,19 @@ describe('formly-form', () => {
>  
>        const listener = sinon.spy()
>        const expression = sinon.spy()
> +      const deepLinkField = getNewField()
> +      deepLinkField.key = 'foo.bar'
> +      deepLinkField.watcher = {
> +        listener: '',
I don't think that this test is testing what you think it is... I think you're getting code coverage, but the watchExpression here never gets evaluated. Could you add a new test that has a deepLink field like this except for the expression uses a spy and asserts that the spy was called properly?

—
Reply to this email directly or view it on GitHub.